deploy/buildtools/buildconfigfile.py

```diff
@@ -0,0 +1,161 @@
+
+        self.args = args
+
+        global_build_config_file = None
+        with open(args.buildconfigfile, "r") as yaml_file:
+            global_build_config_file = yaml.safe_load(yaml_file)
+
+        # aws specific options
+        self.agfistoshare = global_build_config_file['agfis-to-share']
+        self.acctids_to_sharewith = global_build_config_file['share-with-accounts'].values()
+
+        # this is a list of actual builds to run
+        builds_to_run_list = global_build_config_file['builds']
+
+        build_recipes_config_file = None
+        with open(args.buildrecipesconfigfile, "r") as yaml_file:
+            build_recipes_config_file = yaml.safe_load(yaml_file)
+
+        build_farm_hosts_config_file = None
+        with open(args.buildfarmconfigfile, "r") as yaml_file:
+            build_farm_hosts_config_file = yaml.safe_load(yaml_file)
+
+        build_recipes = dict()
+        for section_name, section_dict in build_recipes_config_file.items():
+            if section_name in builds_to_run_list:
+                build_recipes[section_name] = BuildConfig(
+                    section_name,
+                    section_dict,
+                    build_farm_hosts_config_file,
+                    self,
+                    launch_time)
+
+        self.hwdb = RuntimeHWDB(args.hwdbconfigfile)
+
+        self.builds_list = list(build_recipes.values())
+        self.build_ip_set = set()
+
+    def setup(self) -> None:
+        """Setup based on the types of build farm hosts."""
+        for build in self.builds_list:
+            auto_create_bucket(build.s3_bucketname)
+
+        # check to see email notifications can be subscribed
+        get_snsname_arn()
+
+    def request_build_farm_hosts(self) -> None:
+        """Launch an instance for the builds. Exits the program if an IP address is reused."""
+
+        def categorize(seq: List[BuildConfig]) -> Dict[str, List[BuildConfig]]:
+            """Bucket build configs by their build farm host dispatcher name.
+
+            Args:
+                seq: List of build configs to categorize.
+
+            Returns:
+                Dict of lists where the key is the build farm host name and the list holds
+                the build configs associated with it.
+            """
+            class_names = list(map(lambda x: x.build_farm_host_dispatcher.__class__.__name__, seq))
+            class_builds_zipped = zip(class_names, seq)
+
+            d = defaultdict(list)
+            for c, b in class_builds_zipped:
+                d[c].append(b)
+
+            return d
+
+        cat = categorize(self.builds_list)
+
+        for bhd_class, build_farm_hosts in cat.items():
+            # batch launching build farm hosts of similar types
+            getattr(import_module("buildtools.buildfarmhostdispatcher"),
+                bhd_class).request_build_farm_hosts(list(map(lambda x: x.build_farm_host_dispatcher, build_farm_hosts)))
```

Comment:
The least change way to avoid the getattr lookup of the class (since you don't need to call the staticmethod using it) is to:

```suggestion
        for build_farm_hosts in cat.values():
            # batch launching build farm hosts of similar types
            build_farm_hosts[0].request_build_farm_hosts(list(map(lambda x: x.build_farm_host_dispatcher, build_farm_hosts)))
```

Reply:
Closing since this optimization is unnecessary anymore.